connect_ext_test.go

```diff
@@ -2190,8 +2190,8 @@ func TestStreamUnexpectedEOF(t *testing.T) {
 			_, err = responseWriter.Write(payload[:len(payload)-1])
 			assert.Nil(t, err)
 		},
-		expectCode: connect.CodeInvalidArgument,
-		expectMsg:  fmt.Sprintf("invalid_argument: protocol error: promised %d bytes in enveloped message, got %d bytes", len(payload), len(payload)-1),
+		expectCode: connect.CodeInternal,
+		expectMsg:  "internal: incomplete envelope: unexpected EOF",
```

Comment:
Can you check what `grpc-go` does here? I think I changed this to `CodeInvalidArgument` specifically to match them, but I may be wrong.

Reply:
Starts as `io.ErrUnexpectedEOF` and converted to `status.Error(codes.Internal, err.Error())`
- https://github.com/grpc/grpc-go/blob/b0a946cf0cf1adb094d7da9dbf2d054905702c73/rpc_util.go#L619
- https://github.com/grpc/grpc-go/blob/82a568ddbb02ff410a4f7c071ecf8b277992a251/rpc_util.go#L883
Oh I thought this was my wrong doing 😅

Reply:
What direction is this? Returning "internal" from the server, when the _client_ has sent an invalid stream, seems very wrong. Internal should be reserved for internal server errors, not errors in the client (akin to 5xx vs 4xx). You generally want exception reporting/alerting for internal server errors since they indicate a likely bug in your code; you do _not_ want any such alerting for client errors, because then a client can induce alerts in your system w/ carefully crafted (invalid) requests.

So I agree that this should be "internal" only in cases where it's the _server_ that sent the invalid message.